core/templates/pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.component.ts

```diff
@@ -134,12 +134,42 @@ export class TopicsAndSkillsDashboardPageComponent {
     this.directiveSubscriptions.unsubscribe();
   }
 
-  generateNumbersTillRange(range: number): number[] {
-    let arr: number[] = [];
-    for (let i = 0; i < range; i++) {
-      arr.push(i);
+  getPages(current: number, total: number): number[] {
+    /**
+     * This function returns an array of the total number of pages.
+     * "-1" is an indicator to print ellipses.
+     * The "total" variable is being rounded off to avoid cases
```

Comment:
I don't understand this reasoning. If the total passed in is an integer then why round it? On the other hand if the total being passed in is not an integer then it doesn't actually represent the total number of pages and its name is misleading.

Also I suggest that instead of a long JS comment here, you write a JSDoc -- that would be more appropriate for documenting the overall purpose of a function.

Reply:
If totalSkills = 40 and itemsPerPage = 15, then totalNumberOfPages = 40/15 = 2.66. I am using the ceiling function to handle such cases. The ceiling function makes totalNumberOfPages  = 3. On other hand, totalNumberOfPages is an integer when itemsPerPage is 10 or 20.

Can I create the JSDoc  in the "topics-and-skills-dashboard-page" folder or do you have a specific location where you want me to put it? I couldn't find a directory which had all the JSDocs.

Reply:
JSDocs should go directly above their function, i.e. in this file next to the code, not in a separate file. See examples [here](https://jsdoc.app/tags-example.html).

For your first paragraph, something is off with the naming, because the thing you're calling "total" isn't actually the total. (Another red flag here is that you're redefining the variable.) Instead, you might perhaps consider passing in the total number of items and the number of items per page as separate arguments.

Reply:
Thanks a lot, have addressed all the suggested changes. @seanlip